Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpTransceiverBackend.cpp

```diff
@@ -131,40 +131,40 @@ static inline WARN_UNUSED_RETURN ExceptionOr<GstCaps*> toRtpCodecCapability(cons
         for (auto& attribute : codec.sdpFmtpLine.split(';')) {
             auto components = attribute.split('=');
             auto field = makeString(codecName.convertToASCIILowercase(), '-', components[0]);
-            gst_caps_set_simple(caps, field.ascii().data(), G_TYPE_STRING, components[1].ascii().data(), nullptr);
+            gst_caps_set_simple(caps.get(), field.ascii().data(), G_TYPE_STRING, components[1].ascii().data(), nullptr);
         }
     }
 
     if (msid)
-        gst_caps_set_simple(caps, "a-msid", G_TYPE_STRING, msid, nullptr);
+        gst_caps_set_simple(caps.get(), "a-msid", G_TYPE_STRING, msid.characters(), nullptr);
 
-    GST_DEBUG("Codec capability: %" GST_PTR_FORMAT, caps);
+    GST_DEBUG("Codec capability: %" GST_PTR_FORMAT, caps.get());
     return caps;
 }
 
-static GUniquePtr<char> getMsidFromCurrentCodecPreferences(GstWebRTCRTPTransceiver* transceiver)
+static ASCIILiteral getMsidFromCurrentCodecPreferences(GstWebRTCRTPTransceiver* transceiver)
 {
     GRefPtr<GstCaps> currentCaps;
-    GUniquePtr<char> msid;
     g_object_get(transceiver, "codec-preferences", &currentCaps.outPtr(), nullptr);
     GST_TRACE_OBJECT(transceiver, "Current codec preferences: %" GST_PTR_FORMAT, currentCaps.get());
     if (gst_caps_get_size(currentCaps.get()) > 0) {
         auto* s = gst_caps_get_structure(currentCaps.get(), 0);
-        msid = GUniquePtr<char>(g_strdup(gst_structure_get_string(s, "a-msid")));
+        if (auto msIdValue = gstStructureGet<ASCIILiteral>(s, "a-msid"_s))
+            return *msIdValue;
```

Comment:
`StringView` makes no difference in this context because when built from `ASCIILiteral` it just assigns its `const void* m_characters` from the span data. And it's more complex to use because.

> I don't know if this case is correct because I don't know the life-time of these caps. 

The caps belong to the transceiver for which we keep a ref in `GStreamerRtpTransceiverBackend`.